File: php/spec/reporters/terminal.spec.php
<?php declare(strict_types=1);

use Eno\Reporters\Terminal;

describe('Terminal reporter', function() {
  given('_context', function() {
    require(__DIR__ . '/context_fixture.php');
    return $context;
  });

  it('produces text output', function() {
    $this->_context->reporter = new Terminal;

    $snippet = Terminal::report(
      $this->_context,
      $this->_context->instructions[1],
      $this->_context->instructions[0]
    );

    expect($snippet)->toMatchSnapshot('spec/reporters/snapshots/terminal.snap.sh');
  });
});

File: ruby/spec/reporters/terminal.spec.rb
# frozen_string_literal: true

input = <<~DOC.strip
> comment
# section

field: value

field_with_items:
- item
- item

> comment
- item

## subsection

field_with_attributes:
attribute = value

> comment
attribute = value
DOC

describe Enolib::TerminalReporter do
  it 'produces colored terminal output' do
    document = Enolib.parse(input, reporter: Enolib::TerminalReporter)
    context = document.instance_variable_get(:@context)

    reporter = context.reporter.new(context)
    snippet = reporter.report_element(context.document[:elements].first).snippet()

    # Uncomment to inspect
    # puts snippet

    expect(snippet).to match_snapshot
  end
end
